### DISCUSSION TITLE
Bug - Data List Checkbox can't be checked

### DISCUSSION THREAD

**Discussion by nmp1testing** [patternfly/patternfly#6997] [Q&A] [answered]
The checkbox(es) of the [Data List Checkbox example](https://www.patternfly.org/components/data-list#checkboxes-actions-and-additional-cells) can't be checked. Is there a working example of this component?

**Comment by ShivamG0217** [comment]
try adding defaultChecked attribute to the component <DataListCheck>

Below is an example:

`<DataListCheck
            aria-labelledby="check-action-item1"
            name="check-action-check1"
            defaultChecked 
          />`

**Reply by kmcfaul** [reply, marked as answer]
The issue is that `DataListCheck` doesn't handle uncontrolled checkboxes, and examples of are not fully wired up to handle checkbox events. You would need track the checked state in the app state, pass that to `checked` on `DataListCheck` as well as pass an event handler to `onChange` to toggle the checked state.

`<DataListCheck checked={isCheckedState} onChange={onChangeHandler}  aria-labelledby="check-action-item1" name="check-action-check1" />`

There is a [PR open](https://github.com/patternfly/patternfly-react/pull/10916) for `DataList` targeting the next major version (v6) that will resolve the issue with uncontrolled checkboxes not functioning, so the behavior will work by default and you won't need to wire up the example.
